pkg/totalaccountwatcher/totalaccountwatcher.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/go-logr/logr"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/openshift/aws-account-operator/pkg/localmetrics"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+const (
+	AwsSecretName = "aws-account-operator-credentials"
+)
+
+// ErrAwsAccountLimitExceeded indicates the orgnization account limit has been reached.
+var ErrAwsAccountLimitExceeded = errors.New("AccountLimitExceeded")
+
+// TotalAccountWatcher global var for TotalAccountWatcher
+var TotalAccountWatcher *totalAccountWatcher
+
+var log = logf.Log.WithName("aws-account-operator")
+
+type totalAccountWatcher struct {
+	watchInterval time.Duration
+	AwsClient     awsclient.Client
+	client        client.Client
+	Total         int
+}
+
+// Initialize creates a global instance of the TotalAccountWatcher
+func Initialize(client client.Client, watchInterval time.Duration) {
+	log.Info("Initializing the totalAccountWatcher")
+
+	AwsClient, err := awsclient.GetAWSClient(client, awsclient.NewAwsClientInput{
+		SecretName: AwsSecretName,
+		NameSpace:  awsv1alpha1.AccountCrNamespace,
+		AwsRegion:  "us-east-1",
+	})
+
+	if err != nil {
+		log.Error(err, "Failed to get AwsClient")
+		return
+	}
+
+	TotalAccountWatcher = NewTotalAccountWatcher(client, AwsClient, watchInterval)
+	TotalAccountWatcher.UpdateTotalAccounts(log)
+}
+
+// NewTotalAccountWatcher returns a new instance of the TotalAccountWatcher interface
+func NewTotalAccountWatcher(client client.Client, AwsClient awsclient.Client, watchInterval time.Duration) *totalAccountWatcher {
+	return &totalAccountWatcher{
+		watchInterval: watchInterval,
+		AwsClient:     AwsClient,
+		client:        client,
+	}
+}
+
+// TotalAccountWatcher will trigger AwsLimitUpdate every `scanInternal` and only stop if the operator is killed or a
+// message is sent on the stopCh
+func (s *totalAccountWatcher) Start(log logr.Logger, stopCh <-chan struct{}) {
+	log.Info("Starting the totalAccountWatcher")
+	for {
+		select {
+		case <-time.After(s.watchInterval):
+			log.Info("totalAccountWatcher: updating a")
+			err := s.UpdateTotalAccounts(log)
+			if err != nil {
+				log.Error(err, "totalAccountWatcher not started, awsLimit won't be updated")
+			}
+		case <-stopCh:
+			log.Info("Stopping the totalAccountWatcher")
+			break
+		}
+	}
+}
+
+// UpdateTotalAccounts will update the TotalAccountWatcher's total field
+func (s *totalAccountWatcher) UpdateTotalAccounts(log logr.Logger) error {
+
+	// Reset cache
+	secretList := &corev1.SecretList{}
+
+	listOps := &client.ListOptions{Namespace: awsv1alpha1.AccountCrNamespace}
+	if err := TotalAccountWatcher.client.List(context.TODO(), listOps, secretList); err != nil {
+		log.Error(err, fmt.Sprintf("Unable to list secrets in namespace %s", awsv1alpha1.AccountCrNamespace))
+	}
```

Comment:
@liamwazherealso what is happening here? I don't think this is required.

Reply:
This is the code that reloaded the cache. Removing.